FrameworkSystem/scripts/dirac-proxy-destroy.py

```diff
@@ -0,0 +1,181 @@
+
+
+
+
+class Params( object ):
+  """
+  handles input options for dirac-proxy-destroy
+  """
+  def __init__( self ):
+    """
+    creates a Params class with default values
+    """
+    self.vos = []
+    self.delete_all = False
+
+  def addVO( self, voname ):
+    """
+    adds a VO to be deleted from remote proxies
+    """
+    self.vos.append( voname )
+    return S_OK()
+
+  def setDeleteAll( self, _ ):
+    """
+    deletes local and remote proxies
+    """
+    self.delete_all = True
+    return S_OK()
+
+  def needsValidProxy( self ):
+    """
+    returns true if any remote operations are required
+    """
+    return self.vos or self.delete_all
+
+
+  # note the magic : and =
+  def registerCLISwitches( self ):
+    """
+    add options to dirac option parser
+    """
+    Script.setUsageMessage( "Script to delete a dirac proxy. Default: delete local proxy only." )
+    Script.registerSwitch( "a", "all", "Delete the local and all uploaded proxies (the nuclear option)", self.setDeleteAll )
+    Script.registerSwitch( "v:", "vo=", "Delete uploaded proxy for vo name given", self.addVO )
+
+
+
+def getProxyGroups():
+  """
+  Returns a set of all remote proxy groups stored on the dirac server for the user invoking the command.
+  """
+  proxies = gProxyManager.getUserProxiesInfo()
+  if not proxies[ 'OK' ]:
+    raise RuntimeError( 'Could not retrieve uploaded proxy info.' )
+
+
+  user_groups = set()
+  for dn in proxies[ 'Value' ]:
+    dn_groups = set( proxies[ 'Value' ][ dn ].keys() )
+    user_groups.update( dn_groups )
+
+  return user_groups
+
+
+def mapVoToGroups( voname ):
+  """
+  Returns all groups available for a given VO as a set.
+  """
+
+  vo_dict =  Registry.getGroupsForVO( voname )
+  if not vo_dict[ 'OK' ]:
+    raise RuntimeError( 'Could not retrieve groups for vo %s.' % voname )
+
+  return set( vo_dict[ 'Value' ] )
+
+
+
+def deleteRemoteProxy( userdn, vogroup ):
+  """
+  Deletes proxy for a vogroup for the user envoking this function.
+  Returns a list of all deleted proxies (if any).
+  """
+  rpcClient = RPCClient( "Framework/ProxyManager" )
+  retVal = rpcClient.deleteProxyBundle( [ ( userdn, vogroup ) ] )
+
+  if retVal[ 'OK' ]:
+    gLogger.notice( 'Deleted proxy for %s.' % vogroup )
+  else:
+    gLogger.error( 'Failed to delete proxy for %s.' % vogroup )
+
+
+def deleteLocalProxy( proxyLoc ):
+  """
+  Deletes the local proxy.
+  Returns false if no local proxy found.
+  """
+  try:
+    os.unlink( proxyLoc )
+  except IOError:
+    gLogger.error( 'Failed to delete local proxy.' )
+    return
+  gLogger.notice( 'Local proxy deleted.' )
+
+
+
+def main():
+  """
+  main program entry point
+  """
+  options = Params()
+  options.registerCLISwitches()
+
+  Script.parseCommandLine( ignoreErrors = True )
+
+  if options.delete_all and options.vos:
+    gLogger.error( "-a and -v options are mutually exclusive. Please pick one or the other." )
+    sys.exit( 1 )
```

Comment:
it would be better if there are no calls to exit here.
Instead return 1 and call exit in the `__main__` block below.
Calling exit in functions can cause problems when documentation is created, or when calling the code through unittests, etc.